pilot/pkg/xds/proxy_dependencies.go

```diff
@@ -60,9 +60,9 @@ func proxyDependentOnConfig(proxy *model.Proxy, config model.ConfigKey, push *mo
 	// Detailed config dependencies check.
 	switch proxy.Type {
 	case model.SidecarProxy:
-		if proxy.SidecarScope.DependsOnConfig(config) {
+		if proxy.SidecarScope.DependsOnConfig(config, push.Mesh.RootNamespace) {
```

Comment:
just curious, why did we split this out? Not a big deal, just slightly preferred the old approach

Reply:
yeah, a small change just to reduce SidecarScope unnecessary field